src/main/scala/org/globalforestwatch/summarystats/SummaryAnalysisFactory.scala

```diff
@@ -1,76 +1,101 @@
 import org.globalforestwatch.summarystats.carbonflux.CarbonFluxAnalysis
 import org.globalforestwatch.summarystats.carbon_sensitivity.CarbonSensitivityAnalysis
+import org.globalforestwatch.summarystats.firealerts.FireAlertsAnalysis
 import org.globalforestwatch.summarystats.gladalerts.GladAlertsAnalysis
 import org.globalforestwatch.summarystats.treecoverloss.TreeLossAnalysis
 
 case class SummaryAnalysisFactory(analysis: String,
-                                  featureRDD: RDD[Feature[Geometry, FeatureId]],
+                                  featureObj: org.globalforestwatch.features.Feature,
                                   featureType: String,
-                                  part: HashPartitioner,
+                                  featureUris: NonEmptyList[String],
                                   spark: SparkSession,
                                   kwargs: Map[String, Any]) {
 
   val runAnalysis: Unit =
-    analysis match {
-      case "annualupdate" =>
-        AnnualUpdateAnalysis(
-          featureRDD: RDD[Feature[Geometry, FeatureId]],
-          featureType: String,
-          part: HashPartitioner,
-          spark: SparkSession,
-          kwargs: Map[String, Any]
-        )
-      case "annualupdate_minimal" =>
-        AnnualUpdateMinimalAnalysis(
-          featureRDD: RDD[Feature[Geometry, FeatureId]],
-          featureType: String,
-          part: HashPartitioner,
-          spark: SparkSession,
-          kwargs: Map[String, Any]
-        )
-      case "carbonflux" =>
-        CarbonFluxAnalysis(
-          featureRDD: RDD[Feature[Geometry, FeatureId]],
-          featureType: String,
-          part: HashPartitioner,
-          spark: SparkSession,
-          kwargs: Map[String, Any]
-        )
-     case "carbon_sensitivity" =>
-        CarbonSensitivityAnalysis(
-          featureRDD: RDD[Feature[Geometry, FeatureId]],
-          featureType: String,
-          part: HashPartitioner,
-          spark: SparkSession,
-          kwargs: Map[String, Any]
-        )
-      case "gladalerts" =>
-        GladAlertsAnalysis(
-          featureRDD: RDD[Feature[Geometry, FeatureId]],
-          featureType: String,
-          part: HashPartitioner,
-          spark: SparkSession,
-          kwargs: Map[String, Any]
-        )
-      case "treecoverloss" =>
-        TreeLossAnalysis(
-          featureRDD: RDD[Feature[Geometry, FeatureId]],
+    if (analysis != "firealerts") {
+      // ref: https://github.com/databricks/spark-csv
+      val featuresDF: DataFrame =
+        FeatureDF(featureUris, featureObj, kwargs, spark)
+
+      /* Transition from DataFrame to RDD in order to work with GeoTrellis features */
+      val featureRDD: RDD[Feature[Geometry, FeatureId]] =
+        FeatureRDD(featuresDF, featureObj)
+
+      val inputPartitionMultiplier = 64
+
+      val part = new HashPartitioner(
+        partitions = featureRDD.getNumPartitions * inputPartitionMultiplier
+      )
```

Comment:
I am not a fan of this implementation.
Is there a way to keep this out of the factory?
Ie, there could be a featureRDD trait and two subclasses, one that returns the geotrellis features, the other the spatial RDD?
Creation of the the RDD could live in a separate class rddFactory